Apsitvarkom.Api/Controllers/PollutedLocationController.cs

```diff
@@ -131,4 +134,31 @@ public async Task<ActionResult<PollutedLocationResponse>> Create(PollutedLocatio
 
         return CreatedAtAction(nameof(GetById), new { response.Id }, response);
     }
-}
\ No newline at end of file
+
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(List<string>))]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpPut("Update")]
+    public async Task<ActionResult<PollutedLocationResponse>> Update(PollutedLocationUpdateRequest pollutedLocationUpdateRequest)
+    {
+        var validatorResponse = await _pollutedLocationUpdateValidator.ValidateAsync(pollutedLocationUpdateRequest);
+        if (!validatorResponse.IsValid) return BadRequest(validatorResponse.Errors.Select(e => e.ErrorMessage).ToList());
+
+        var existingLocation = await _repository.GetByPropertyAsync(x => x.Id == pollutedLocationUpdateRequest.Id);
+        if (existingLocation is null) return StatusCode(StatusCodes.Status500InternalServerError);
+       
+        var mappedLocation = _mapper.Map<PollutedLocationUpdateRequest, PollutedLocation>(pollutedLocationUpdateRequest, existingLocation);
+        if (mappedLocation is null) return StatusCode(StatusCodes.Status500InternalServerError);
+
+        try
+        {
+            await _repository.UpdateAsync(mappedLocation);
+
+            return Ok(mappedLocation);
```

Comment:
`mappedLocation` is of type `PollutedLocation`, however in the method signature you state, that you will return `ActionResult<PollutedLocationResponse>`. we need to map this once again for the types to match.
Take a look on how this can be achieved [here](https://github.com/vu-vibedosa/apsitvarkom-be/blob/staging/Apsitvarkom.Api/Controllers/PollutedLocationController.cs#L138)

Reply:
Note: Do this before the UpdateAsync call, so that in case we can't return proper results and fail, the data wouldn't be changed as wel..